packages/execution-context/README.md

```diff
@@ -0,0 +1,79 @@
+# @clipboard-health/execution-context <!-- omit from toc -->
+
+`ExecutionContext` is a lightweight Node.js package built with TypeScript that leverages `AsyncLocalStorage` to create a statically available context parallel to any execution. It provides a reliable, thread-safe context for attaching and accessing metadata throughout the lifecycle of an execution, such as API requests, background jobs, or message consumers. This allows various parts of your application to communicate and share metadata without needing to explicitly pass context objects.
+
+### Features
+
+- **Scoped Contexts**: Attach data to a context that is specific to each execution scope.
+- **Static Access**: Access context data statically anywhere in the codebase within an active execution.
+- **Metadata Aggregation**: Store and retrieve metadata across the lifespan of an execution, ideal for logging, tracing, and other observability tasks.
+
+## Table of Contents <!-- omit from toc -->
+
+- [Install](#install)
+- [Usage](#usage)
+- [Local development commands](#local-development-commands)
+
+## Install
+
+```bash
+npm install @clipboard-health/execution-context
+```
+
+## Usage
+
+This example demonstrates how to create a logging context, accumulate metadata from various function calls, and then log a single message containing all the gathered metadata.
+
+```typescript
```

Comment:
Typically, we add an `examples` folder and then use [embedme](https://github.com/zakhenry/embedme) to embed the example into the REAMDE. That way, the example stays up-to-date with the library and type checks.

In the README, you put:

```
<!-- prettier-ignore -->
```ts
// ./examples/YOUR_EXAMPLE.ts
```

In package.json:

```json
"scripts": {
  "embed": "embedme README.md"
}
```

Then run `npm run embed` from the root directory. CI ensures the README and example files match so they don't get out of sync.